### DISCUSSION TITLE
release more platform and arch executable file

### DISCUSSION THREAD

**Discussion by axetroy** [denoland/deno#10020] [General]
https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-1

According to rust's documentation, it can support `linux/arm64` build.

Is there any problem to build `linux/arm64`? (V8 or something else)

In my opinion, there will be more and more processors with arm64 architecture.

| Platform/Arch  | Real Machine |
| ------------- | ------------- |
| windows/arm64  | `Surface Pro X`  |
| linux/arm64  | `Pinebook`  |
| linux/mips64  | `-`  |
| linux/mips64le  | `-`  |
| linux/ppc64  | `-`  |
| linux/ppc64le  | `-`  |

All of the above platforms and archs, rust all supports.

I think it will be great to supports multiple platforms.

**Comment by kitsonk** [comment]
There is a cost and overhead to building for many platforms. Currently we are building the M1 builds on our personal machines, as GitHub is still working on support.

Also we would obviously want to ensure all the dependencies, including v8, support the architecture. Especially with some of the io/fs stuff there can be all sorts of "gotchas".

Cross compiling isn't the approach we would likely want as well.

So this is a good suggestion, but more complicated than flipping a switch.

**Reply by jheysonsaav** [reply]
> There is a cost and overhead to building for many platforms. Currently we are building the M1 builds on our personal machines, as GitHub is still working on support.
> 
> Also we would obviously want to ensure all the dependencies, including v8, support the architecture. Especially with some of the io/fs stuff there can be all sorts of "gotchas".
> 
> Cross compiling isn't the approach we would likely want as well.
> 
> So this is a good suggestion, but more complicated than flipping a switch.

@kitsonk In a personal project that I have (also written in rust), I have created this github actions https://github.com/wondwise/wondwise/blob/main/.github/workflows/deploy.yml workflow to compile for multiple platforms, maybe what can be a problem is v8

**Comment by Dygear** [comment]
Ideally Apple M1 ([`aarch64-apple-darwin`](https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-2)) would be supported out of the box, as well as the Raspberry Pi 4's Broadcom BCM2711B0 ([`aarch64-unknown-linux-gnu`](https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-1)) triple targets. Getting a 32bit ARM version of V8 is turning out to be VERY, VERY difficult to support, but these two targets can be compiled out of the box on a machine with a decent amount of RAM, 8GB in our case, with a 2-4GB swap. It takes time on the Raspberry Pi, but it does absolutely produce a usable artifact. So it's a matter of making this work at scale in some way. There is a whole discussion about this going on in the issue tracker. https://github.com/denoland/deno/issues/1846 https://github.com/denoland/deno/issues/4862

**Reply by kitsonk** [reply]
> Apple M1 (aarch64-apple-darwin) 

We already support it and it is available, it is just that GitHub doesn't yet have build servers, so we build it manually and upload it, so users only get released versions and no "canary".

**Reply by Dygear** [reply]
> > Apple M1 (aarch64-apple-darwin)
> 
> We already support it and it is available, it is just that GitHub doesn't yet have build servers, so we build it manually and upload it, so users only get released versions and no "canary".

Yeah, that was more of a ... "These are the platforms I would like to specifically target."

**Comment by bnoordhuis** [comment]
MIPS is arguably a dying architecture. I wrote or merged the occasional fix to make Node.js build on that architecture but it was sometimes broken for months without anyone noticing. On the other hand, Loongson seems to be maintaining the MIPS port of V8 pretty well.

The PPC market is nearly all IBM clients. I worked at IBM until ~6 months ago. Pretty much everyone who uses Node.js on ppc64 pays IBM in one way or another.

V8's ppc64 port is well maintained, with several full-time maintainers. ppc64be is exclusively AIX. ppc32 is dead.
